modules/services/osm_wikibase.js

```diff
@@ -0,0 +1,185 @@
+import _debounce from 'lodash-es/debounce';
+import _forEach from 'lodash-es/forEach';
+
+import { json as d3_json } from 'd3-request';
+
+import { utilQsString } from '../util';
+
+
+var apibase = 'https://wiki.openstreetmap.org/w/api.php';
+var _inflight = {};
+var _wikibaseCache = {};
+
+
+var debouncedRequest = _debounce(request, 500, { leading: false });
+
+function request(url, callback) {
+    if (_inflight[url]) return;
+
+    _inflight[url] = d3_json(url, function (err, data) {
+        delete _inflight[url];
+        callback(err, data);
+    });
+}
+
+
+export default {
+
+    init: function() {
+        _inflight = {};
+        _wikibaseCache = {};
+    },
+
+
+    reset: function() {
+        _forEach(_inflight, function(req) { req.abort(); });
+        _inflight = {};
+    },
+
+
+    /** List of data items representing language regions.
+     *  To regenerate, use Sophox query:  http://tinyurl.com/y6v9ne2c (every instance of Q6999)
```

Comment:
This seems like a list that will grow over time, perhaps even as frequently as once per release cycle. Consider splitting it out into a separate file (JSON perhaps) that can be updated as part of iD’s build process.

Reply:
Why can't the service just fetch these on `init()` ?

Reply:
Does `init()` support async?  In any case, this would be a fast-enough call to do on the first request to the service, and cache it.  Patch is on its way...

Reply:
The taginfo interface does a similar lookup on `init()` to fetch popular keys.  Just try to make sure that if it fails iD will still work ok (maybe just without documentation lookups)

https://github.com/openstreetmap/iD/blob/e51f2eb1e706e369d8e8518818d6ee5747a9cf15/modules/services/taginfo.js#L202-L208